### ISSUE TITLE
[com.paypal.android.p2pmobile] PayPal Crash

### ISSUE THREAD

**Issue by Slayer5934** [microg/GmsCore#1421]
**Affected app**
Name: PayPal
Package id: com.paypal.android.p2pmobile

**Describe the bug**
Starting the app and logging in results in a crash after it attempts to "Download English language files."

**System**
Android Version: 10
Custom ROM: Havoc-OS version 3.11 on a Moto X4.

**microG**
microG Core version: 0.2.17.204714
microG Self-Check results: All ticked.

**Additional context**
Crash Log: https://del.dog/lekilanedu.txt

**Comment by mar-v-in** [comment]
Do you have Google Play Store installed? What is the source of the APK?

**Comment by Slayer5934** [comment]
Aurora Store is the source, do not have Play Store installed.

**Comment by MrNessuno** [comment]
I can login without problems, I have MicroG core 0.2.18 and Android 11

**Comment by Axolord** [comment]
As noted in Issue #1435, it seems to be a problem with the newest Paypal app (version >7.38.1). Version 7.39.0 did not work for me (microG v 0.2.18.204714, Android 10, Havoc OS), while previous versions work fine.

**Comment by TheLastProject** [comment]
I can confirm this with com.paypal.android.p2pmobile version Version 7.42.3 on Android 10 with microG 0.2.19.211515, having the app installed from Aurora Store.

My crash log is as follows:
```
FATAL EXCEPTION: main
Process: com.paypal.android.p2pmobile, PID: 22176
com.google.android.play.core.tasks.RuntimeExecutionException: com.google.android.play.core.internal.ar: Failed to bind to the service.
	at o.gzx.a(Unknown Source:16)
	at o.lkw$g.e(Unknown Source:252)
	at o.gzj.run(Unknown Source:23)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)
Caused by: com.google.android.play.core.internal.ar: Failed to bind to the service.
	at o.gtp.e(Unknown Source:80)
	at o.gti.c(Unknown Source:4)
	at o.gtd.run(Unknown Source:0)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.os.HandlerThread.run(HandlerThread.java:67)
```

Hope this helps

**Comment by drrossum** [comment]
Have you tried the latest version of microg already?

**Comment by TheLastProject** [comment]
I managed to update to 0.2.21.212158 (most up to date in F-Droid microG repo), sadly the error message remains the exact same

**Comment by TheLastProject** [comment]
I have upgraded PayPal to 8.0.1 and no longer have any crashes. So the crash itself seems to no longer by triggered by more recent PayPal versions.

**Comment by Slayer5934** [comment]
Should be fine then, someone else can comment here if the issue still happens and I'll re-open.

**Comment by MrNessuno** [comment]
The last working version is 8.11, newer version will crash at startup

**Comment by TheLastProject** [comment]
Version 8.14.2 indeed crashes on my Android 11 (LineageOS for microG) with microG 0.2.24.214816 with the absolute weirdest crash:

```
FATAL EXCEPTION: main
Process: com.paypal.android.p2pmobile, PID: 30271
java.lang.RuntimeException: Unable to create application o.syb: java.lang.ArithmeticException: divide by zero
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6724)
	at android.app.ActivityThread.access$1300(ActivityThread.java:237)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7664)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.ArithmeticException: divide by zero
	at o.xjm.h(Unknown Source:500)
	at o.yfv.a(:54)
	at o.xjm.b(:75)
	at com.paypal.android.p2pmobile.common.app.BasePayPalApplication.k(:2438)
	at com.paypal.android.p2pmobile.common.app.BasePayPalApplication.K(:835)
	at com.paypal.android.p2pmobile.common.app.BasePayPalApplication.onCreate(:615)
	at o.szg.onCreate(:43)
	at o.syb.onCreate(:8)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6719)
	... 8 more
```

The thing is though, I recognize no reference to play so if this is microG related it most likely has to do with microG returning an empty list somewhere and PayPal trying to use that later. Debugging this would probably not be easy at all.

**Comment by PiJ82** [comment]
> The last working version is 8.11, newer version will crash at startup

I just checked, the last working version I can confirm is 8.12.3.

Microg 0.2.24.214816.

**Comment by TheLastProject** [comment]
Moved my crash to a new issue given this one is closed to make is obvious to the maintainers the issue is not resolved: #1710

**Comment by ale5000-git** [comment]
If anyone still has issues please open a new ticket.
